### PR TITLE
Setup issue label notifier

### LINKED ISSUE #9216: Issue Label Notifications by using GitHub Action
### Motivation

As PyTorch Lightning Community grows, some users want to be notified by issues with specific labels. By using GitHub Action would allow to notify the interested users. 
Basically, the GitHub Action runs every time a label is attached to an issue, and compares the label to the list of notification recipients that has been specified If a match is found, the action will add a comment to the issue and @mention the notification recipients.

### Pitch

Introduce Issue label notifier by using GitHub Action. 
https://github.com/marketplace/actions/issue-label-notifier

### Alternatives
Keep it as it is. 
Use the UI search tool to find the issues/PR with corresponding labels. 

### Additional context

Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/9140

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #9216 
Blocked by #10473

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/notifier.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  notify:
+    runs-on: ubuntu-latest
+    steps:
+        - uses: jenschelkopf/issue-label-notification-action@1.3
+          with:
+             recipients: |
+                  help wanted=@akihironitta
+                  documentation=@akihironitta
```
Comment (carmocca) [inline .github/workflows/notifier.yml:15]: I had in mind copying what's done in https://github.com/pytorch/pytorch/issues/24422

Otherwise, it is too much of a hassle to subscribe/unsubscribe (needs 3 reviews + CI).

If the action you chose does not support it, we should try and implement it.

**Comment by akihironitta** [comment]
Closing this PR in favour of pytorch-probot
